src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -98,35 +126,161 @@ public CourseScoresDTO calculateCourseScores(long courseId, Collection<Long> stu
             }
             plagiarismCases = plagiarismCaseRepository.findByCourseId(courseId);
         }
-        var plagiarismMapping = PlagiarismMapping.createFromPlagiarismCases(plagiarismCases);
-        double finalMaxPointsInCourse = maxPointsInCourse; // needed to make the variable effectively final for lambda below
-        double finalReachableMaxPointsInCourse = reachableMaxPointsInCourse; // needed to make the variable effectively final for lambda below
-        var studentScores = studentIdToParticipations.entrySet().parallelStream()
-                .map(entry -> calculateCourseScoreForStudent(entry.getKey(), entry.getValue(), finalMaxPointsInCourse, finalReachableMaxPointsInCourse, plagiarismMapping))
-                .toList();
-        var course = courseExercises.iterator().next().getCourseViaExerciseGroupOrCourseMember();
-        return new CourseScoresDTO(maxPointsInCourse, reachableMaxPointsInCourse, course.getPresentationScore(), studentScores);
+
+        Course course = courseExercises.iterator().next().getCourseViaExerciseGroupOrCourseMember();
+
+        List<StudentScoresForExamBonusSourceDTO> studentScoresForExamBonusSource = studentIdToParticipations.entrySet().parallelStream()
+                .map(entry -> constructStudentScoresForExamBonusSourceDTO(course, entry, maxAndReachablePoints, plagiarismCases)).toList();
+
+        return new CourseScoresForExamBonusSourceDTO(maxAndReachablePoints.maxPoints, maxAndReachablePoints.reachablePoints, course.getPresentationScore(),
+                studentScoresForExamBonusSource);
+    }
+
+    private StudentScoresForExamBonusSourceDTO constructStudentScoresForExamBonusSourceDTO(Course course, Map.Entry<Long, List<StudentParticipation>> entry,
```

Comment:
I think this would be more readable if the Long and the List were two different parameters. Then you can use `participations.isEmpty()` below instead of `entry.getValue().isEmpty()`.